ShapeManager.cpp

```diff
@@ -0,0 +1,16 @@
+#ifndef ShapeManager_cpp
+#define ShapeManager_cpp
+
+#include "Area.cpp"
+#include "Volume.cpp"
+#include "Input.cpp"
+#include "Shape.cpp"
+
+class ShapeManager : public Area, public Volume, public Input, public Shape 
+// This class follows the Single Responsibility Principle. 
```

Comment:
The name / description of the responsibility of the class do not match.

The class name suggests, it *manages* `Shape`s. However, from the code, we can see that `ShapeManager` is a `Shape` itself.